platform/engineering/design-docs/2021-08-01-autonomous-deployments.md

```diff
@@ -0,0 +1,211 @@
+
+We will also require commits to be merged into `staging` (not `dev`) before they are allowed to merge into `master`. After this requirement is in place, we would like to monitor for feedback to see if we need to allow teams to bypass this requirement with branch naming `hotfix/` and/or `revert/` prefixes.
+#### Roadmap
+
+1. Update current system to function with new workflows
+2. Create and release staging/dev deployment behaviors
+3. Create and release initial production behaviors with "allow" list
+4. Create and release MVP production behaviors with "disallow" list
+
+#### Technologies Being Used
+
+- [Github Actions](https://github.com/features/actions)
+- [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/Welcome.html)
+- [Webpack](https://webpack.js.org/)
+
+### Work Estimates
+
+#### Milestone: Update current system to function with new workflows
+
+- Update current (dev/staging) deploy scripts to sync and not delete files from s3 OR create new script
+- Create single app build in webpack that includes assets.
+
+This work should be completed in 1 sprint with 2 people (1 person on each task).
+
+#### Milestone: Create and release staging/dev deployment behaviors
+
+- Create deployment script for dev
+- Disable current dev deployment behaviors
+- Announce and reserve time for troubleshooting dev before moving to staging
+- Create deployment script for staging
+- Disable current staging deployment behaviors
+- Reserve time for troubleshooting staging
+- Reach out to two application teams for production deployment beta
+
+This work should be completed in 2-3 sprints with 2 people. Sprint 1 should be dev. Sprint 2 should be staging. Reserving and extra sprint for troubleshooting if needed.
+
+#### Milestone: Create and release initial production behaviors with "allow" list
+
+- Create "allow" list for applications
+- Create production deployment workflow
+- Monitor auto and manual deployments for a week to debug and troubleshoot
+
+This work should be completed in 2-3 sprints with 2 people, one of which should be very familiar with Github Actions.
+
+#### Milestone: Create and release MVP production behaviors with "disallow" list
+
+- Update deployment workflow to use disallow list instead of allow list
+- Plan and execute VFS training on the update
+- Release to teams
+- Monitor auto and manual deployments for a sprint to debug and troubleshoot
+
+This work should be completed in 2-3 sprints with 2 people.
+
+### Alternatives
+
+An alternative that was discussed was isolating the applications in their own repositories before executing autonomous deploys. We felt that this would bloat the timeline and delay the feature from being released for too long.
+
+We also discussed monorepo isolation, but deemed it as unnecessary to complete the objective.
+
```

Comment:
Any other alternative details? Would be great to fill this section out a bit more to show that we've done our due diligence.